enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,362 @@
+communicates this fact and provides instructions for canceling the operation if
+the user wishes to back out of the operation.  This enhancement applies to the
+default IngressController as well as custom IngressControllers.
+
+### Goals
+
+1. The cluster administrator can change the scope of the service load-balancer of an IngressController with "LoadBalancerService" endpoint publishing strategy type.
+2. If the cloud platform supports mutating scope on a service load-balancer, the operator will perform this operation when the cluster administrator changes the configured scope.
+3. If the cloud platform does not support mutating scope, the operator will advise the user how to complete the operation.
+
+### Non-Goals
+
+1. The operator does not delete a service load-balancer for an IngressController that is not marked for deletion.
+
+## Proposal
+
+Some platforms (such as Azure and GCP) support changing the scope of a service
+load-balancer between internal and external without deleting and recreating the
+load balancer, by setting [a cloud-provider-specific
+annotation](https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer)
+on the Kubernetes Service object.  On these platforms, the operator merely sets
+the annotation to the desired scope, and Kubernetes's service controller and
+cloud-provider implementation complete the operation of changing the load
+balancer's scope.
+
+Other platforms (such as AWS) require deleting and recreating a load balancer to
+change its scope.  This operation is disruptive: It interrupts ingress traffic
+and may cause the load balancer's address to change.  On these platforms, the
+operator signals that the user must delete the Kubernetes Service object.  Once
+the user performs this step, the operator recreates the load balancer with the
+desired scope to complete the operation.
+
+In order to signal that the user must take action, the operator sets the
+`Progressing` status condition to `True` with a message that indicates the
+action that the user must take:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: custom
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  domain: mycluster.com
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+status:
+  availableReplicas: 2
+  conditions:
+  - message: Deployment has minimum availability.
+    reason: MinimumReplicasAvailable
+    status: "True"
+    type: Available
+  - message: The IngressController scope was changed from "External" to "Internal".  To put this change into effect, you must delete the openshift-ingress/router-default service; the service load-balancer will then be deprovisioned and a new one created.  Alternatively, you can change the IngressController's spec.endpointPublishingStrategy.loadBalancer.scope field value back to its previous value.
```

Comment:
Can we also emit oc commands that you can simply cut & paste? I'm assuming we have enough context (e.g., namespace and names, et al). Might be helpful should you want to restore "its previous value" too.

Reply:
We can emit oc commands, but the patch command alone makes the message too long to pass the markdownlint CI job.  I'll try using double quotes with a backslash-escaped newline to see whether the linter allows that.  I believe that it'll still be valid yaml, although it won't be exactly what `oc get ... -o yaml` would actually output.